pages/weather/index.js

```diff
@@ -0,0 +1,108 @@
+import Layout from "../../components/Layout.js";
+import { useState } from "react";
+import styled from "styled-components";
+
+const api = {
+  key: "0a1b4079174262c9bd54e2c2800cda16",
+  base: "https://api.openweathermap.org/data/2.5/",
+};
+export default function Weather() {
+  const [query, setQuery] = useState("");
+  const [weather, setWeather] = useState({});
+  const searchPressed = () => {
+    fetch(`${api.base}weather?q=${query}&units=metric&APPID=${api.key}`)
+      .then((response) => response.json())
+      .then((result) => {
+        console.log(result);
+        setWeather(result);
+      });
+  };
+
+  const dateBuilder = (d) => {
+    let months = [
+      "January",
+      "February",
+      "March",
+      "April",
+      "May",
+      "June",
+      "July",
+      "August",
+      "September",
+      "October",
+      "November",
+      "December",
+    ];
+    let days = [
+      "Sunday",
+      "Monday",
+      "Tuesday",
+      "Wednesday",
+      "Thursday",
+      "Friday",
+      "Saturday",
+    ];
+    let day = days[d.getDay()];
+    let date = d.getDate();
+    let month = months[d.getMonth()];
+    let year = d.getFullYear();
+    return `${day} ${date} ${month} ${year}`;
+  };
```

Comment:
This looks very cumbersome, I've found [the `toLocaleDateString` method which might be helpful](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString#using_options), but it uses commas between the values.